tests/console/irqbalance.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Package: irqbalance
+# Summary: Test irqbalance runs as service as well as standalone
+# Maintainer: Pavel Dostál <pdostal@suse.cz>
+
+use warnings;
+use base "consoletest";
+use strict;
+use testapi qw(is_serial_terminal :DEFAULT);
+use utils qw(systemctl zypper_call clear_console);
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Return when balancing is ineffective on system with a single cpu
+    zypper_call("in coreutils");
+    my $nproc = script_output('nproc');
+    die('Balancing is ineffective on systems with a single cpu') unless($nproc>1);
+
+    zypper_call("in irqbalance");
+    systemctl('enable --now irqbalance.service');
+
+    # Generate CPU load
+    script_run('dd if=/dev/urandom of=/dev/null count=6 bs=16M iflag=fullblock', 90);
```

Comment:
Is this really enough load for checking the interrupt distribution? While it takes roughly 2 seconds on my laptop to complete, I'd increase it to a safe value, like `count=30` to ensure this will also work on more capable machines (e.g. bare metal workers with horsepower CPUs).